pkg/controller/operators/catalog/operator.go

```diff
@@ -499,3 +505,19 @@ func (o *Operator) getSourcesSnapshot(plan *v1alpha1.InstallPlan, includedNamesp
 
 	return sourcesSnapshot
 }
+
+// getExistingCSVNames returns a set of existing CSV names for the given namespace
+func (o *Operator) getExistingCSVNames(namespace string) (map[string]struct{}, error) {
+	// Get a list of CSV CRs in the namespace
+	csvList, err := o.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).List(metav1.ListOptions{Watch: false})
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a set of CSV names from the list
+	existing := make(map[string]struct{})
```

Comment:
There's an edge case with this (and the consumer side in `resolveCRDDescription`).  There can be more than one CSV in a namespace that owns a CRD for a couple of reasons:

- Could be in the middle of upgrading
- Could have attempted to create one and failed (so status is failed but owner is still there)
- Anyone with `csv` `create` permission can write arbitrary owned CRDs (they won't be accepted, but they can write them).

The simplest way to fix this would be to just fail the installplan if we see two csvs that own the CRD already in the namespace (we can't tell which to resolve). Ideally we would just retry the resolution after a bit, and only mark the installplan as truly "Failed" if we've tried a few times and never got a single owner for a CRD (which indicates a bad namespace state - something on the CSV side got stuck).

We could try to detect the upgrading states of the CSVs and watch them directly, but I'd rather keep things more decoupled between catalog/olm.

Anyway this is return a list of existing CSVs in the namespace, but I think what you actually want is a map from a CRDName to the CSVName in the namespace that owns it (then you don't have to loop in `resolveCRDDescription`